[Desmond Yong Shao Tian] iP

desmondyst opened the pull request:
### DukePro

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ *SUPER* FAST to use

All you need to do is,
1. download it from [here](https://nus-cs2103-ay2223s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3.  add your tasks.
4. let it manage your tasks for you 😉 

And it is **FREE**!

Features:

- [X] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

___

If you Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

desmondyst committed:
Complete Level 1 - (Greet, Echo, Exit)

desmondyst committed:
Complete Level 2 - (Add, List)

desmondyst committed:
Complete Level 3 - (Mark as done)

desmondyst committed:
Complete Level 4 - (ToDos, Events, Deadlines)

desmondyst committed:
Completed -  Automated Text UI Testing

desmondyst committed:
Completed Level 5 - (Handle Errors)

desmondyst committed:
Completed Level 6 - (Delete)

desmondyst committed:
Completed - (Use Enums)

desmondyst committed:
Fixed ACTUAL.TXT different from  EXPECTED.TXT for A-TextUiTesting

desmondyst committed:
Added saving and loading the tasks to hard disk - (Level 7)

desmondyst committed:
Add Dates and Times - (Level 8)

desmondyst committed:
Merge branch 'branch-Level-7'

desmondyst committed:
Merge branch 'branch-Level-8'

desmondyst committed:
Add More OOP - (A - MoreOOP)

desmondyst committed:
Organize into Packages

desmondyst committed:
Merge remote-tracking branch 'add-gradle-support/add-gradle-support'

desmondyst committed:
Modify Gradle configuration

desmondyst committed:
Add JUnit tests

desmondyst committed:
Add JavaDoc comments

desmondyst committed:
Create a JAR File

desmondyst committed:
Merge branch 'branch-A-JavaDoc'

desmondyst committed:
Tweak code to comply with coding standard

desmondyst committed:
Add feature that allows user to find a task by searching for a keyword

desmondyst committed:
Merge branch 'branch-A-CodingStandard'

desmondyst committed:
Merge branch 'branch-Level-9'

desmondyst committed:
Use CheckStyle

desmondyst committed:
Merge branch 'branch-A-CheckStyle'

desmondyst committed:
Add a GUI to Duke

desmondyst committed:
Merge branch 'branch-Level-10'

desmondyst committed:
Use Varargs

desmondyst committed:
Merge branch 'branch-A-Varargs'

qiaoen17 commented on src/main/java/duke/command/Command.java:
```diff
@@ -7,7 +7,7 @@
 public abstract class Command {
     /**
-     * Perform an action due to the command.
+     * Performs an action due to the command.
```
I liked how you changed your JavaDocs method headers to start with Performs instead of Perform, I noticed it in other JavaDocs too.

qiaoen17 commented on src/main/java/duke/command/DeleteCommand.java:
```diff
@@ -10,7 +10,7 @@
     private TasksList tasksList;
     private String[] inputArray;
-    private static final String deleteMsg = "Noted. I've removed this task:\n";
+    private static final String DELETE_MSG = "Noted. I've removed this task:\n";
```
I like that you took note of changing `deleteMsg` to `DELETE_MSG`. I noticed you have changed it for the other constants too

qiaoen17 commented on src/main/java/duke/command/EventCommand.java:
```diff
@@ -5,7 +5,7 @@
 
 /**
- * Represents a command to add a Event to the tasksList.
+ * Represents a command to add an Event to the tasksList.
```
I like that you took note of grammar in your JavaDocs!

qiaoen17 commented on src/main/java/duke/task/Task.java:
```diff
@@ -68,26 +68,25 @@ public static Task fromStorageString(String storageString) {
-                break;
-            default:
-                throw new RuntimeException(String.format("Invalid task type %s.", taskType));
+        case TODO:
+            task = new Todo(storageStringArray[2]);
+            break;
+        case EVENT:
+            try {
+                task = new Event(storageStringArray[2], storageStringArray[3]);
+            } catch (DukeException exception) {
+                throw new RuntimeException(exception.getMessage());
+            }
+            break;
+        case DEADLINE:
+            try {
+                task = new Deadline(storageStringArray[2], storageStringArray[3]);
+            } catch (DukeException exception) {
+                throw new RuntimeException(exception.getMessage());
+            }
+            break;
+        default:
+            throw new RuntimeException(String.format("Invalid task type %s.", taskType));
```
I like that the indentation for switch cases was followed. I think you took note of the coding standards and followed them strictly.

qiaoen17 reviewed (commented):
Looks good :)

limweijun commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,76 @@
+     * Creates a new storage instance from the given pathname string.
+     * @param pathname A pathname string.
+     */
+    public Storage(String pathname) {
```
the variable pathname should be pathName to follow the java coding standard (variable names should be written in camelCase)

desmondyst committed:
The classes in the application do not have assertions to verify the
assumption that the Tasks and the List of Tasks is never null.

This make the program susceptible to null pointer exception.

Let's add assertions to the program to ensure that the runtime can
verify this assumption and flag them out if this assumption does not
hold, which is, when the Tasks and List of Tasks is null.

Classes affected:
* Command
  * DeadlineCommand
  * DeleteCommand
  * EventCommand
  * FindCommand
  * ListCommand
  * MarkCommand
  * ToDoCommand
* Task
  *Task
  *TasksList

desmondyst committed:
Nesting of multiple try-catch block inside switch cases.

Let's have one single try-catch block for the entire switch statement
instead.

This improves code quality and readability.

desmondyst committed:
No way to update existing task.

Troublesome to remove and add task just to update.

Let's add a command for the user to update a current existing
task.

desmondyst committed:
Add JavaDoc comments

desmondyst committed:
Merge pull request #1 from desmondyst/branch-A-Assertions

Add Assertions

desmondyst committed:
Merge branch 'master' of https://github.com/desmondyst/ip

desmondyst committed:
Merge branch 'master' into branch-A-CodeQuality

desmondyst committed:
Merge branch 'master' into branch-C-Update

# Conflicts:
#	src/main/java/duke/task/TasksList.java

desmondyst committed:
Merge pull request #2 from desmondyst/branch-A-CodeQuality

Improve code quality

desmondyst committed:
Merge branch 'master' of https://github.com/desmondyst/ip

desmondyst committed:
Merge branch 'master' into branch-C-Update

desmondyst committed:
Merge pull request #3 from desmondyst/branch-C-Update

Add ability to update task

desmondyst committed:
Merge branch 'master' of https://github.com/desmondyst/ip

desmondyst committed:
Create gradle.yml

Set up CI

desmondyst committed:
Fix styling

desmondyst committed:
Add user guide

desmondyst committed:
Update git ignore to stop tracking data files

desmondyst committed:
Fix styling issues for user guide
